### PR TITLE
monitor_search is not yet supported

### PR DESCRIPTION
**Pull request by unclebconnor**
removing `monitor_search` flag while unsupported

### CODE REVIEW

File: website/docs/r/service_level_objective.html.markdown [hunk 1]
```diff
@@ -96,17 +96,16 @@ The following options are specific to the `type` of service level objective:
...
     numerator   = "sum:my.custom.count.metric{type:good}.as_count()"
-    denominator = "sum:my.custom.count.metric{*}.as_count()" 
+    denominator = "sum:my.custom.count.metric{*}.as_count()"
 }
 ```
 
 * `monitor` type SLOs:
-    * `monitor_ids` - (Optional) A list of numeric monitor IDs for which to use as SLIs. Their tags will be auto-imported into `monitor_tags` field in the API resource. At least 1 of `monitor_ids` or `monitor_search` must be provided.
-    * `monitor_search` - (Optional) The monitor query search used on the monitor search API to add monitor_ids by searching. Their tags will be auto-imported into `monitor_tags` field in the API resource. At least 1 of `monitor_ids` or `monitor_search` must be provided.
+    * `monitor_ids` - (Optional) A list of numeric monitor IDs for which to use as SLIs. Their tags will be auto-imported into `monitor_tags` field in the API resource.
```
Comment (bkabrda) [inline website/docs/r/service_level_objective.html.markdown]: If one of `monitor_ids` and `monitor_search` must be provided and we are hiding `monitor_search`, shouldn't we then also make `monitor_ids` required for now?

Comment (unclebconnor) [reply website/docs/r/service_level_objective.html.markdown]: @bkabrda that makes sense to me.  I haven't actually updated any code here...should we just document that `monitor_ids` is required for now?

Comment (unclebconnor) [reply website/docs/r/service_level_objective.html.markdown]: Adjusted so `monitor_ids` is required

**Comment by platinummonkey** [comment]
can you remove this from the schema as well?

File: website/docs/r/service_level_objective.html.markdown (hunk 1 above)
Comment (unclebconnor) [reply website/docs/r/service_level_objective.html.markdown]: @bkabrda this should be ready for another review when you can

**Comment by bkabrda** [comment]
Hmm, I'm honestly unsure about this. The Terraform deprecation policy [1] prevents us from just removing an attribute like this. OTOH this attribute never worked anyway, so I guess this should be fine...

I think after all I'm 👍 for doing this change, but would like to hear more opinions. @nmuesch thoughts?

[1] https://www.terraform.io/docs/extend/best-practices/deprecations.html#provider-attribute-removal

**Comment by platinummonkey** [comment]
since it's not in use, I think it's safe to just drop.

**Comment by unclebconnor** [comment]
@bkabrda @nmuesch any additional thoughts here?  ok to merge/drop the setting since it's not in use?  Other changes?

**Comment by bkabrda** [comment]
One last question that I have here is if there's a specific timeframe on where `monitor_search` will be actually supported. IOW, is it worth removing it?

**Comment by unclebconnor** [comment]
@bkabrda that option is currently not production ready and there is a good bit of design and dev work that needs to be done and hasn't been scheduled.  We've already had a support case where someone was trying to use it and getting errors so we'd prefer to remove it.

**Comment by bkabrda** [comment]
@unclebconnor ok, that's fair, let's remove the field for now.

I was going through the terraform docs for removal of attribute [1]. The "right" way to do this seems to be to leave the field there, but add a `Removed` attribute to it, e.g. `Removed: "We identified problems with usage of this field and thus we're marking it as 'removed' until these are fixed"` or something similar. This will make sure that our users who tried to define this field will get a proper error message when running terraform and will know what's going on.

[1] https://www.terraform.io/docs/extend/best-practices/deprecations.html#provider-attribute-removal

**Comment by unclebconnor** [comment]
@bkabrda Ok I've made that update.  I'm pretty sure this is the format that is described in the docs but let me know if you have other feedback

**Comment by unclebconnor** [comment]
@bkabrda not sure of process for this repo but I'm assuming you want to do the final merge.  Let me know if there's anything else you need from me here.

**Comment by bkabrda** [comment]
LGTM now, merging. 👍 thanks for your patience :)

**Pull request merged**